src/Model/Join.php

```diff
@@ -167,30 +167,39 @@ protected function init(): void
     {
         $this->_init();
 
-        // handle foreign table containing a dot
+        // owner model should have id_field set
+        $id_field = $this->getOwner()->id_field;
+        if (!$id_field) {
+            throw (new Exception('Joins owner model should have id_field set'))
+                ->addMoreInfo('model', $this->getOwner());
+        }
+
+        // handle foreign table containing a dot - that will be reverse join
         if (is_string($this->foreign_table) && strpos($this->foreign_table, '.') !== false) {
+            // split by LAST dot in foreign_table name
+            [$this->foreign_table, $this->foreign_field] = preg_split('/\.+(?=[^\.]+$)/', $this->foreign_table);
```

Comment:
```suggestion
            [$this->foreign_table, $this->foreign_field] = preg_split('~\.+(?=[^.]+$)~', $this->foreign_table);
```

if I get the regex correctly...